tests/optional_mutex/main.c

```diff
@@ -0,0 +1,125 @@
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "optional_mutex.h"
+#include "warnp.h"
+
+#define NUM_TIMES 100
+
+static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
As far as I know, it's safe to do this even if we don't `-lpthread`.  I mean, the header is required to be present; I don't see anything like [XSI] to indicate that it's optional in POSIX.  So it should be able to compile it, regardless of whether any pthread symbols are used (or are available) in the link stage.

But I acknowledge that it looks and feels weird.